src/features/ImeiProgramming/index.tsx

```diff
@@ -0,0 +1,433 @@
+
+                                        await writeIMEI(
+                                            device.serialNumber,
+                                            useCloud ? cloudIMEI : manualIMEI
+                                        )
+                                            .then(() => {
+                                                setStatus('FINISHED');
+                                            })
+                                            .catch(() => {
+                                                setError(
+                                                    'Failed to write IMEI.'
+                                                );
+                                                if (!hasProgrammedFirmware) {
+                                                    setStatus('FIRMWARE');
+                                                } else {
+                                                    setStatus('UNSUPPORTED');
+                                                }
+                                            });
+                                    });
+                                }}
+                            >
+                                Write IMEI
+                            </DialogButton>
+                        )}
+                        <DialogButton
+                            variant="secondary"
+                            disabled={showSpinner}
+                            onClick={() => setStatus('NONE')}
+                        >
+                            Close
+                        </DialogButton>
+                    </>
+                }
+            >
+                {error && (
+                    <div className="tw-mb-4">
+                        <div className="tw-text-gray-70 tw-flex tw-flex-row tw-items-center tw-gap-4 tw-bg-gray-50 tw-p-4">
+                            <span className="mdi mdi-alert tw-align-middle tw-text-2xl/6 tw-text-red" />
+                            <div>{error}</div>
+                        </div>
+                    </div>
+                )}
+                {status === 'INIT' && (
+                    <div>Checking that IMEI can be written to device.</div>
+                )}
+                {status === 'PROTECTED' && (
+                    <div>
+                        Recover the device in order to check whether programming
+                        IMEI is supported.
+                    </div>
+                )}
+                {status === 'UNSUPPORTED' && (
+                    <div>
+                        Programming IMEI is not supported for this device.
+                        {cloudIMEI && <CopyImeiNotice imei={cloudIMEI} />}
+                    </div>
+                )}
+                {status === 'FIRMWARE' && (
+                    <>
+                        <div>
+                            Unable to communicate with the device. Do you want
+                            to program ptm firmware?
```

Comment:
```suggestion
                            to program PTM firmware?
```
Smells like an acronym...